Blish HUD/Controls/_Types/UndoRedoStack.cs

```diff
@@ -0,0 +1,73 @@
+﻿/*
+ *  This code is heavily adapted from the Myra UndoRedoStack (https://github.com/rds1983/Myra/blob/a9dbf7a1ceedc19f9e416c754eaf38e89a89a746/src/Myra/Graphics2D/UI/TextEdit/UndoRedoStack.cs)
+ *
+ *  MIT License
+ *
+ *  Copyright (c) 2017-2020 The Myra Team
+ *
+ *  Permission is hereby granted, free of charge, to any person obtaining a copy
+ *  of this software and associated documentation files (the "Software"), to deal
+ *  in the Software without restriction, including without limitation the rights
+ *  to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ *  copies of the Software, and to permit persons to whom the Software is
+ *  furnished to do so, subject to the following conditions:
+ *
+ *  The above copyright notice and this permission notice shall be included in all
+ *  copies or substantial portions of the Software.
+
+ *  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ *  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ *  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ *  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ *  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ *  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ *  SOFTWARE.
+ */
+
+using System.Collections.Generic;
+
+namespace Blish_HUD.Controls {
+    internal class UndoRedoStack {
+
+        private readonly Stack<UndoRedoRecord> _stack = new Stack<UndoRedoRecord>();
+
+        public Stack<UndoRedoRecord> Stack => _stack;
```

Comment:
Maybe return a copy of the stack so the stack cannot be altered outside of this class (you could also just hope that no one in this project will do smth like that)

Reply:
The `UndoRedo` method in `TextInputBase` needed access to directly pop it.  It's been marked as internal and nothing else within the assembly should be poking at it too much anyways (I'd hope we'd notice anything weird in a PR).

That being said, I realized the calls in `UndoRedo` that need direct access to the stack could probably get reimplemented directly into `UndoRedoStack` and exposed with a single function call, so I've gone ahead and done that and removed the property for the Stack.

Reply:
Looks good!